store/tikv/region_request.go

```diff
@@ -167,7 +168,19 @@ func (s *RegionRequestSender) sendReqToRegion(bo *Backoffer, ctx *RPCContext, re
 	if e := tikvrpc.SetContext(req, ctx.Meta, ctx.Peer); e != nil {
 		return nil, false, errors.Trace(e)
 	}
-	resp, err = s.client.SendRequest(bo.ctx, ctx.Addr, req, timeout)
+	// judge the store limit switch.
+	if atomic.LoadUint32(&config.GetGlobalConfig().StoreLimit) > 0 {
+		if err := s.getStoreToken(ctx.Store); err != nil {
+			logutil.BgLogger().Warn("get store token failed", zap.Error(err))
+			return nil, false, errors.Trace(err)
+		}
+		resp, err = s.client.SendRequest(bo.ctx, ctx.Addr, req, timeout)
+		if err := s.releaseStoreToken(ctx.Store); err != nil {
```

Comment:
If there's a panic in `SendRequest`, the token won't be released. So I suggest deferring `releaseStoreToken`, to make sure it will always release.